src/inmanta/env.py

```diff
@@ -455,6 +456,8 @@ def run_pip_install_command_from_config(
                 del sub_env["PIP_INDEX_URL"]
             if "PIP_PRE" in sub_env:
                 del sub_env["PIP_PRE"]
+            if "PIP_NO_INDEX" in sub_env:
+                del sub_env["PIP_PRE"]
```

Comment:
Nice catch! As this is a bugfix, I think we need to add a bugfix section for this to the changelog.

Reply:
I think this was actually intentional. We only ignore the environment variables for things that can be set via the PipConfig. `--no-index` cannot be set using the `PipConfig` so we don't remove the environment variable when `use_system_config` is False.

Reply:
As discussed this will stay and I'll add a bugfix section

Reply:
```suggestion
            if "PIP_NO_INDEX" in sub_env:
                del sub_env["PIP_NO_INDEX"]
```